pandas/core/algorithms.py

```diff
@@ -104,6 +104,12 @@ def _ensure_data(values, dtype=None):
             dtype = values.dtype
         else:
             # Datetime
+            if values.ndim > 1:
```

Comment:
what exactly hits this?

Reply:
`DataFrame.rank` with all-datetime64 columns.  #27015 has a terrible terrible [hack](https://github.com/pandas-dev/pandas/pull/27015/files?file-filters%5B%5D=.py#diff-03b380f521c43cf003207b0711bac67fR8372) instead of this 5-line workaround,.

Reply:
we haven't reviewed #27015 not averse, just want to avoid hacks on hacks here

Reply:
Yah, this is distinctly the non-hack solution.